packages/ecs/src/ComponentFunctions.ts

```diff
@@ -561,61 +563,78 @@ export const setComponent = <C extends Component>(
   component.onSet(entity, component.stateMap[entity]!, args)
 
   const propagate = () => {
-    const layer = EntityLayerState.getEntityLayer(entity)
-    for (const dstLayerID of Object.keys(layer.relations)) {
-      if (layer.relations[dstLayerID] === 'propagate') {
-        const dstEntity = EntityLayerState.getLinkedEntity(entity, dstLayerID as LayerID)
-
-        // if (!entityExists(dstEntity)) {
-        //if the linked entity doesn't exist, we recreate it and sync its state with the source entity
-        // dstEntity = createEntity(dstLayerID as LayerID)
-        // EntityLayerState.linkEntity(entity, dstEntity)
-        // for (const entityComponent of getAllComponents(entity)) {
-        //   if (entityComponent === component) continue //we're about to do this operation anyways
-        //   setComponent(dstEntity, entityComponent, entityComponent.stateMap[entity]!.get(NO_PROXY_STEALTH))
-        // }
-        // }
-
-        //switch any target entities in the args to their corresponding linked entity in the destination layer
-        if (component.schema) {
-          const componentSchema = component.schema as TTypedSchema<C>
-          const frontier = [{ schema: componentSchema, setArgs: args }] as { schema: any; setArgs: any }[]
-          while (frontier.length > 0) {
-            const { schema, setArgs } = frontier.pop()!
-            if (!schema || typeof setArgs !== 'object') continue
-            for (const key in setArgs) {
-              const valSchema = schema.properties?.[key] as any
-              //check if the value is an entity
-              if (
-                valSchema?.properties?.options &&
-                valSchema.properties.options['id'] === 'Entity' &&
-                setArgs[key] !== UndefinedEntity
-              ) {
-                //if so, we need to switch it to the linked entity in the destination layer
-                const linkedEntity = EntityLayerState.getLinkedEntity(setArgs[key], dstLayerID as LayerID)
-                setArgs[key] = linkedEntity
-              } else if (typeof setArgs[key] === 'object') {
-                frontier.push({ schema: valSchema, setArgs: setArgs[key] })
-              }
-            }
+    if (!component.jsonID) return
+    if (component === LayerComponent || LayerComponents.includes(component as any)) return
+    if (LayerComponent.get(entity) !== Layers.Authoring) return
+    if (component === UUIDComponent) return
+
+    const uuidComponent = getOptionalComponent(entity, UUIDComponent)
+    if (!uuidComponent) return
+    const linkedEntity = UUIDComponent.getEntityByUUID(uuidComponent, Layers.Simulation)
```

Comment:
Need to re-implement the enforcement of arbitrary layer relations instead of hard-coding propagation as authoring -> simulation